test/parallel/test-cli-syntax.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const assert = require('assert');
+const spawnSync = require('child_process').spawnSync;
+const path = require('path');
+
+const common = require('../common');
+
+var node = process.argv[0];
+
+// test both sets of arguments that check syntax
+var syntax_args = [
+  ['-c'],
+  ['--check']
+];
+
+// test good syntax with and without shebang
+[
+  'syntax/good_syntax.js',
+  'syntax/good_syntax',
+  'syntax/good_syntax_shebang.js',
+  'syntax/good_syntax_shebang',
+].forEach(function(file) {
+  file = path.join(common.fixturesDir, file);
+
+  // loop each possible option, `-c` or `--check`
+  syntax_args.forEach(function(args) {
+    var _args = args.concat([file]);
+    var c = spawnSync(node, _args, {encoding: 'utf8'});
+
+    // no output should be produced
+    assert.equal(c.stdout, '', 'stdout produced');
+    assert.equal(c.stderr, '', 'stderr produced');
+    assert.equal(c.status, 0, 'code == ' + c.status);
+  });
+});
+
+// test bad syntax with and without shebang
+[
+  'syntax/bad_syntax.js',
+  'syntax/bad_syntax',
+  'syntax/bad_syntax_shebang.js',
+  'syntax/bad_syntax_shebang'
+].forEach(function(file) {
+  file = path.join(common.fixturesDir, file);
+
+  // loop each possible option, `-c` or `--check`
+  syntax_args.forEach(function(args) {
+    var _args = args.concat([file]);
+    var c = spawnSync(node, _args, {encoding: 'utf8'});
+
+    // no output should be produced
+    assert.equal(c.stdout, '', 'stdout produced');
+    assert(c.stderr, 'stderr not produced');
```

Comment:
The error messages look like this

```
$ ./out/Release/iojs -c ./bad.js
var foo bar;
        ^^^

SyntaxError: Unexpected identifier
    at startup (node.js:99:13)
    at node.js:961:3
```

Checking them would be difficult, because they have line numbers that are subject to change in the future.  We *could* use regex to test them, but that seems unstable and like it could break in the future.